src/NServiceBus.Core/Persistence/InMemory/SagaPersister/InMemorySagaPersister.cs

```diff
@@ -100,14 +96,9 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                     sagaEntity.ConcurrencyCheck(sagaData, version);
                 }
 
-                data.AddOrUpdate(sagaData.Id, id => new VersionedSagaEntity
-                {
-                    SagaData = DeepClone(sagaData)
-                }, (id, original) => new VersionedSagaEntity
-                {
-                    SagaData = DeepClone(sagaData),
-                    VersionCache = original.VersionCache
-                });
+                data.AddOrUpdate(sagaData.Id,
+                    id => new VersionedSagaEntity(DeepClone(sagaData)),
+                    (id, original) => new VersionedSagaEntity(DeepClone(sagaData), original));
```

Comment:
since we always DeepClone when creating a new `VersionedSagaEntity`, should we do that in the ctor instead?